pkg/apis/flux/v1alpha1/validation/fluxconfig.go

```diff
@@ -122,3 +123,26 @@ func ValidateKustomization(kustomization *fluxv1alpha1.Kustomization, fldPath *f
 
 	return allErrs
 }
+
+// ValidateAdditionalSecretResources validates additionalResources
+func ValidateAdditionalSecretResources(additionalResources []fluxv1alpha1.AdditionalResource, shoot *gardencorev1beta1.Shoot, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	if len(additionalResources) == 0 {
+		return allErrs
+	}
+	resources := map[string]gardencorev1beta1.NamedResourceReference{}
+	for _, resource := range shoot.Spec.Resources {
+		resources[resource.Name] = resource
+	}
+	for i, r := range additionalResources {
```

Comment:
Can you validate that `r.TargetName` is a valid k8s object name? 
Probably we can use `apivalidation.NameIsDNSSubdomain`.